lib/uniffi-zcash-cli/src/diff.rs

```diff
@@ -0,0 +1,108 @@
+use clap::ColorChoice;
+use public_api::{diff::PublicApiDiff, tokens::Token, PublicApi};
+use std::path::Path;
+
+use self::{
+    colors::{color_item_with_diff, color_token_stream},
+    git::init_libs_repo,
+    grep_item::GrepItem,
+    utils::{copy_if_not_exists, get_manifest_from_package_name},
+};
+
+const LIBS_REPO_LATEST_FOLDER: &str = "/tmp/librustzcash-diff/latest";
+const LIBS_REPO_CURRENT_FOLDER: &str = "/tmp/librustzcash-diff/current";
+
+mod colors;
+mod git;
+mod grep_item;
+mod grep_type;
+mod utils;
+
+/// The only public function in this module.
+/// Looks for public API difference between two versions of a certain library.
+/// Then based on that difference, greps a given codebase for possible usage
+/// of a changed item of that librarie's public API.
+pub fn generate_diff(
+    package_name: String,
+    package_new_version: String,
+    package_old_version: String,
+    grep_dir: String,
```

Comment:
all those could be `&str`. In general prefer taking `&str` in arguments rather than `String` unless you need an ownership